src/index.ts

```diff
@@ -82,6 +95,69 @@ function tryParseFloat(value: string) {
 	return result;
 }
 
+function normalizeNumber(value: string, allowDot: boolean = true): string {
+	const match = numberPrefixPattern.exec(value);
+	if (!match) {
+		throw new Error('Invalid number: ' + value);
+	}
+	const [, sign, dot] = match;
+	let dots = dot ? 1 : 0;
+	const endingZero = /0$/.test(value);
+	if (sign === '+') {
+		value = value.substring(1);
+	}
+	if (dot === '.') {
+		if (!allowDot) {
+			throw new Error('Invalid number (too many dots): ' + value);
+		}
+		if (sign === '-') {
+			value = '-0' + value.substring(1);
+		} else {
+			value = '0' + value;
+		}
+	} else if (endingZero || !allowDot) {
+		dots = countDots(value);
+	}
+	if (dots > 0) {
+			if (!allowDot) {
+				throw new Error('Invalid number (too many dots): ' + value);
+			}
+			if (endingZero) {
+				value = value.replace(/\.?0+$/, '');
+			}
+	}
+	return value;
+}
+
+function tryParseStrict(value: string): number {
+	const nval = normalizeNumber(value);
+	const result = parseFloat(nval);
+	if (result.toString() !== nval) {
+		if (verifyZero(value) || verifyENotation(value)) {
```

Comment:
Honestly, I think if we just inlined `Math.abs(value) === 0` here it would be pretty clear.

Reply:
do you mean `Math.abs(parseFloat(value)) === 0`?

Reply:
No I don't mean that. Did you have issues w/o the `parseFloat`? Because I wasn't having issues in my console.